src/debug/di/shimstackwalk.cpp

```diff
@@ -1663,6 +1663,11 @@ HRESULT ShimChain::GetStackRange(CORDB_ADDRESS * pStart, CORDB_ADDRESS * pEnd)
         {
             *pEnd = PTR_TO_CORDB_ADDRESS(m_fpRoot.GetSPValue());
         }
+
+        if (m_fpRoot.GetSPValue() < CORDbgGetSP(&m_context))
+        {
+            hr = E_FAIL;
```

Comment:
We probably should return at around line 1653 to avoid side effect to the arguments.
We probably should return a distinct error code to make future diagnostics easier.